orchestra/tests/test_task_lifecycle.py

```diff
@@ -172,117 +177,246 @@ def test_worker_has_reviewer_status(self):
         self.assertTrue(worker_has_reviewer_status(self.workers[6]))
 
     def test_assign_task(self):
```

Comment:
sanity checks:
- are you still covering all of the assignment functionality?
- there's a lot of code deleted in this test.  did we delete any semantically-relevant conditions that we want to put back in?